tests/Test.ADAL.NET.Unit/WsTrustTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.WsTrust;
+using System.IO;
+using System.Xml.Linq;
+
+namespace Test.ADAL.NET.Unit
+{
+    [TestClass]
+    [DeploymentItem("WsTrustResponse.xml")]
+    public class WsTrustTests
+    {
+        [TestMethod]
+        [TestCategory("WsTrustTests")]
```

Comment:
This file under test has a single t:RequestedSecurityToken, but in your code you have a foreach that deals with the case of multiple t:RequestedSecurityToken. It would be good for a test to also look at this case.

Reply:
Hmm, if you are referring to [line 153 of this PR](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1087/files#diff-546f2adc268992cd2106ba3513ae23e2R153), that line:

> foreach (var node in requestedSecurityToken **.Nodes()** )

actually deals with the case of multiple nodes inside ONE "RequestedSecurityToken". (I'm sorry if that trailing `.Nodes()` part might not be considered obvious.)

Besides, the test sample has been around [for 2 years](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/commits/v3.17.3/tests/Test.ADAL.NET.Unit/WsTrustResponse.xml), and I believe it was modeled from real world case.

So, do we still need to have extra test sample here?